src/components/GoalsComponents/MyGoal/MyGoal.tsx

```diff
@@ -36,15 +40,24 @@ const MyGoal: React.FC<MyGoalProps> = ({ goal, dragAttributes, dragListeners })
   const darkModeStatus = useRecoilValue(darkModeState);
   const showPartnerMode = useRecoilValue(displayPartnerMode);
   const subGoalHistory = useRecoilValue(goalsHistory);
-
   const redirect = (state: object, isDropdown = false) => {
     if (isDropdown) {
       navigate(`/MyGoals/${parentGoal?.id || "root"}/${goal.id}?showOptions=true`, { state });
     } else {
       navigate(`/MyGoals/${goal.id}`, { state });
     }
   };
+
   const handleGoalClick = () => {
+    const link = extractLinks(goal.title);
```

Comment:
const url = extractLinks(goal.title);
  if (url) {
    const finalUrl = url.startsWith("http://") || url.startsWith("https://") ? url : "https://" + url;
    window.open(finalUrl, "_blank");
  }
  try this